### PR TITLE
docs: Update Redis connection string instructions for Railway

### PR DESCRIPTION
**Pull request by qbitdoge**
### What
Updated the Railway Redis and Postgres connection string instructions:
- Changed `DATABASE_URL` to use `${{Postgres.RAILWAY_PRIVATE_DOMAIN}}`
- Updated `REDIS_URL` to use `?family=0` to support both IPv6 and IPv4 connections.
- Adjusted instructions to reflect the new `REDIS_URL` configuration

### Why
Railway [does not support IPv4 private networking](https://docs.railway.com/reference/private-networking#caveats), so users need clearer guidance to enable IPv6. Additionally, the egress fee note ensures users are aware of [potential costs](https://docs.railway.com/reference/pricing/plans#resource-usage-pricing) when using public endpoints.

### How
Edited the relevant section in the documentation for clarity and conciseness.

### Testing
Changes are text-only and have been reviewed for accuracy.

### CODE REVIEW

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Referring to [this documentation](https://docs.railway.com/guides/private-networking#known-configuration-requirements-for-ipv6), it says setting `family=0` supports both IPv6 and IPv4 connections, so I think we should set it to that. Then we don't need to add the note about `Redis.REDIS_PUBLIC_URL`

Another thing: the `REDIS_URL` is set in two places (for worker and server environments). Can you update it there?

Also, it would be great if you can update `DATABASE_URL` as well to be `${{Postgres.RAILWAY_PRIVATE_DOMAIN}}` as this Railway change (regarding fees) seem to apply to it as well.

**Comment by qbitdoge** [comment]
@shahednasser thanks so much for the feedback and suggestions. 
I’ve addressed all the changes as requested.
Let me know if there’s anything else that needs adjustment.

File: www/apps/resources/app/deployment/medusa-application/railway/page.mdx [hunk 1]
```diff
@@ -240,6 +240,7 @@ Where:
 - `ADMIN_CORS`'s value is the URL of the admin dashboard. If you don’t have it yet, or you’re deploying the admin with the Medusa application, you can skip adding it for now.
 - `AUTH_CORS`'s value is the URLs of any application authenticating users, customers, or other actor types, such as the storefront and admin URLs. The URLs are separated by commas. If you don’t have the URLs yet, you can set its value later.
 - Change `DISABLE_MEDUSA_ADMIN` to `false` if you’re hosting the admin with the Medusa application.
+- `REDIS_URL`'s value is automatically generated using Railway’s template syntax. Please note that Railway does not support IPv4 private networking, so make sure to add the `?family=0` to the connection string to support both IPv6 and IPv4 connections.
```
Comment (shahednasser) [inline www/apps/resources/app/deployment/medusa-application/railway/page.mdx]:
```suggestion
- `REDIS_URL`'s value is automatically generated using Railway’s template syntax. You add the `?family=0` to the connection string to support both IPv6 and IPv4 connections.
```

Comment (shahednasser) [reply www/apps/resources/app/deployment/medusa-application/railway/page.mdx]: Also can you add the same line in the section below for worker mode environment variables?

**Review by shahednasser — CHANGES_REQUESTED** [review]

**Comment by qbitdoge** [comment]
@shahednasser done! anything else?

**Review by shahednasser — APPROVED** [review]
LGTM thanks!

**Comment by waptik** [comment]
Putting this here as there was some error with postgres. 
`DATABASE_URL` value should be changed to `${{Postgres.DATABASE_URL}}` instead of `${{Postgres.RAILWAY_PRIVATE_DOMAIN}}` because its value remains `postgres.railway.internal` instead of `postgresql://postgres:xxxxxx@postgres.railway.internal:5432/railway`
Another PR must be created to fix it

**Comment by shahednasser** [comment]
You're correct @waptik , when I used it in Railway's dashboard it showed me that the value was the full address but it wasn't. I'll open another PR to make the update

**Comment by waptik** [comment]
> You're correct @waptik , when I used it in Railway's dashboard it showed me that the value was the full address but it wasn't. I'll open another PR to make the update

Can you turn your working architecture into template because i followed the guide and my server instance has been crashing because of `index.html`
image

My variables
```
ADMIN_CORS="https://${{RAILWAY_PUBLIC_DOMAIN}},https://${{RAILWAY_PRIVATE_DOMAIN}}"
AUTH_CORS="https://${{RAILWAY_PUBLIC_DOMAIN}},https://${{RAILWAY_PRIVATE_DOMAIN}}"
STORE_CORS="http://localhost:8000,https://docs.medusajs.com"
DATABASE_URL="${{Postgres.DATABASE_URL}}"
REDIS_URL="${{Redis.REDIS_URL}}?family=0"
JWT_SECRET="****************"
COOKIE_SECRET="****************"
PAYSTACK_API_KEY="****************"
NODE_ENV="production"
PORT="9000"
MEDUSA_ADMIN_EMAIL="me@email.xyz"
MEDUSA_ADMIN_PASSWORD="****************"
MEDUSA_ADMIN_ONBOARDING_TYPE="default"
RAILWAY_PUBLIC_DOMAIN_VALUE="https://${{RAILWAY_PUBLIC_DOMAIN}}"
DISABLE_MEDUSA_ADMIN="false"
MEDUSA_WORKER_MODE="server"
```

**Comment by shahednasser** [comment]
Thank you for your feedback, I would double check that you've followed every step, and if the error is still happening you can open an issue describing your error

**Pull request merged**